client/src/pages/Analytics/components/CompareGraph.tsx

```diff
@@ -0,0 +1,288 @@
+  };
+
+  const getComparisonInfo = () => {
+    const month1String = new Date(
+      monthsArray[month1Index].year,
+      monthsArray[month1Index].month,
+    ).toLocaleString('default', {
+      month: 'short',
+    });
+    const month2String = new Date(
+      monthsArray[month2Index].year,
+      monthsArray[month2Index].month,
+    ).toLocaleString('default', {
+      month: 'short',
+    });
+
+    const month1Avg = getAverage(data1);
+    const month2Avg = getAverage(data2);
+    const percentageChange = (
+      ((Number(month2Avg) - Number(month1Avg)) / Number(month1Avg)) *
+      100
+    ).toFixed(2);
+    const isDecrease = month1Avg > month2Avg;
+
+    return (
+      <CenteredBox
+        style={{
+          flexDirection: 'column',
+        }}
+      >
+        <Typography variant="body1" style={{ width: 157 }}>
+          {month1String} average: {month1Avg}
+        </Typography>
+        <Typography variant="body1" style={{ width: 157 }}>
+          {month2String} average: {month2Avg}
+        </Typography>
+
+        {isDecrease ? (
+          <CenteredBox
+            style={{
+              flexDirection: 'row',
+            }}
+          >
+            <ArrowDownward sx={{ color: 'green' }} />
+            <Typography variant="body1">{percentageChange}%</Typography>
+          </CenteredBox>
+        ) : (
+          <CenteredBox
+            style={{
+              flexDirection: 'row',
+            }}
+          >
+            <ArrowUpward sx={{ color: 'red' }} />
+            <Typography variant="body1">{percentageChange}%</Typography>
+          </CenteredBox>
+        )}
+      </CenteredBox>
+    );
+  };
+
+  const changeMonth1 = (newVal: number) => {
+    setMonth1Index(newVal);
+
+    // TODO set data1 to be the response from fetchDailyExpenses
+    setData1(dailyTotals1);
+    // fetchDailyExpenses(newVal);
+  };
+
+  const changeMonth2 = (newVal: number) => {
+    setMonth2Index(newVal);
+
+    // TODO set data2 to be the response from fetchDailyExpenses
+    setData2(dailyTotals2);
+    // fetchDailyExpenses(newVal);
+  };
+
+  // const fetchDailyExpenses = async (index: number) => {
+  //   // const obj = monthsArray[index];
+  //   // const dateStart = new Date(obj.year, obj.month, 1, 0, 0, 0);
+  //   // const dateEnd =
+  //   //   obj.month == 11
+  //   //     ? new Date(obj.year, obj.month, 31, 23, 59, 59)
+  //   //     : new Date(obj.year, obj.month + 1, 0, 23, 59, 59);
+
+  //   // TODO send a request and return the corresponding response
+  //   return;
+  // };
+
+  return (
+    <BackgroundBox style={{ width: '100%' }}>
+      <CenteredBox>
+        <TextField
+          select
+          id="month1"
+          label="Month 1"
+          size="small"
+          value={month1Index}
+          onChange={(e) => changeMonth1(e.target.value as unknown as number)}
+          style={{ margin: '0 2vw', marginTop: '1vh' }}
+          InputLabelProps={{
+            style: { color: '#C353DB' },
```

Comment:
did we remove hard-coded colors?

Reply:
the colors we use in this page are not part of the global theme as they wont be used anywhere else, so I had to hard code the colors